VDF.Core/ScanEngine.cs

```diff
@@ -199,21 +201,63 @@ public sealed class ScanEngine {
 
 					
 					if (entry.IsImage && entry.grayBytes.Count == 0)
-						GetGrayBytesFromImage(entry);
+						GetGrayBytesFromImage(entry, grayScaleWidth);
 					else if (!entry.IsImage)
-						FfmpegEngine.GetGrayBytesFromVideo(entry, positionList);
+						FfmpegEngine.GetGrayBytesFromVideo(entry, positionList, grayScaleWidth);
 
 					IncrementProgress(entry.Path);
 				});
 			}
 			catch (OperationCanceledException) { }
 		}
 
-		void ScanForDuplicates() {
+		Dictionary<double, byte[]?> createFlippedGrayBytes(FileEntry entry)
+		{
+			var flippedGrayBytes = new Dictionary<double, byte[]?>();
+			if (entry.IsImage)
+				flippedGrayBytes.Add(0, GrayBytesUtils.FlipGrayScale(entry.grayBytes[0]!, grayScaleWidth));
+			else
+				for (int j = 0; j < positionList.Count; j++)
+				{
+					double idx = entry.GetGrayBytesIndex(positionList[j]);
+					flippedGrayBytes.Add(idx, GrayBytesUtils.FlipGrayScale(entry.grayBytes[idx]!, grayScaleWidth));
+				}
 
-			var percentageDifference = 1.0f - Settings.Percent / 100f;
-			var duplicateDict = new Dictionary<string, DuplicateItem>();
+			return flippedGrayBytes;
+		}
+
+		[MethodImpl(MethodImplOptions.AggressiveInlining)]
```

Comment:
I don't think this will be inlined because the method body is too big. AggressiveInlining is just a suggestion for the compiler, you cannot enforce it.

Reply:
Was this only a hint or should it be removed.
(As I added it, I used PercentageDifference as a reference. But is this function small enough to get inlined?)